### PR TITLE
Add support for external JARs

### PR DESCRIPTION
**Pull request by mlutze**
Adds and uses `AddJar` and `RemJar` calls.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
RFR? Also note that I cannot merge this and publish before a new stable version of Flix is released. Just FYI.

**Comment by mlutze** [comment]
Yes, RFR

File: server/src/engine/queue.ts [hunk 1]
```diff
@@ -173,6 +174,8 @@ export async function processQueue () {
       } else if(job.request == jobs.Request.apiAddPkg && !job.src) {
         const base64 = fs.readFileSync(fileURLToPath(job.uri!)).toString('base64')
         socket.sendMessage({...job, base64 })
+      } else if (job.request === jobs.Request.apiAddJar) {
```
Comment (magnus-madsen) [inline server/src/engine/queue.ts:177]: What is this and why only for add?

**Review by magnus-madsen — COMMENTED** [review]
Looks reasonable. Does it work?

File: server/src/engine/queue.ts (hunk 1 above)
Comment (mlutze) [reply server/src/engine/queue.ts:177]: I think it's actually unneeded since addJar does not have an extra payload.

**Comment by mlutze** [comment]
> Looks reasonable. Does it work?

Yes!

Screenshot from 2021-11-12 23-05-50

**Pull request merged**